### PR TITLE
Implement update_descriptor_sets on Vulkan plus corollary changes.

### PR DESCRIPTION
**Pull request by AlphaModder**
This just updates `Device::update_descriptor_sets` to use the new system of views, and implements all currently supported descriptor types. This also meant adding an API for texel buffers.

Some related bits of code (match statements, enums) related to the changes have been reordered/renamed to be consistent. This can be undone if it's an inconvenience.

### CODE REVIEW

File: src/backend/vulkan/src/device.rs [hunk 1]
```diff
@@ -713,6 +713,22 @@ impl d::Device<B> for Device {
         })
     }
 
+    fn create_texel_buffer_view(
+        &mut self, view: &n::BufferView, format: format::Format
+    ) -> Result<n::TexelBufferView, buffer::ViewError> {
+        Ok(n::TexelBufferView { raw: 
+            match self.raw.0.create_buffer_view(vk::BufferViewCreateInfo {
```
Comment (kvark) [inline src/backend/vulkan/src/device.rs]: hmm, where are the match arms?

File: src/core/src/device.rs [hunk 2]
```diff
@@ -187,6 +187,11 @@ pub trait Device<B: Backend>: Clone {
     ) -> Result<B::BufferView, buffer::ViewError>;
 
     ///
+    fn create_texel_buffer_view(
```
Comment (kvark) [inline src/core/src/device.rs]: We are trying to make it fully Vulkan compatible here (at the core API level). Vulkan doesn't have a concept of `TexelBufferView`, it just has `BufferView` and `ImageView`. I believe what you are trying to achieve is in fact our `BufferView`

**Review by kvark — CHANGES_REQUESTED** [review]
Unfortunately, there are issues with the code that prevent me from merging it right away.

File: src/core/src/device.rs (hunk 2 above)
Comment (AlphaModder) [reply src/core/src/device.rs]: Vulkan's `VkBufferView` is indeed equivalent to my `TexelBufferView`. However, in Vulkan, BufferViews are a subsection of a buffer plus a format, and are used exclusively for Texel Buffers. However, the usage of gfx's `BufferView` in`pso::DescriptorWrite::StorageBuffer` and `UniformBuffer` indicates that you intended it to be simply a subsection of a buffer, which has no particular name in Vulkan and does not need to be explicitly created, so I created a new type.

File: src/backend/vulkan/src/device.rs (hunk 1 above)
Comment (AlphaModder) [reply src/backend/vulkan/src/device.rs]: Whoops, that is not a match statement anymore.

**Comment by kvark** [comment]
I see, thanks for explanation! Let's proceed with exactly the way buffer views are exposed in Vulkan

**Comment by AlphaModder** [comment]
What's happened here? It seems that somehow our repositories have become out of sync. I'm noticing that two of your commits have appeared on this PR, with a different ID than the one listed in the main repository.

**Comment by AlphaModder** [comment]
Since the view refactor has already been merged, I am rebasing my changes onto gfx/master and making a PR there.

**Pull request closed without merging**